Fix app upgrade for releave/0.9.18

janherich opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #4077 

### Steps to test:
Try to install the build over `0.9.17` release build

status: ready

janherich committed:
Fix app upgrade for releave/0.9.18

yenda commented on src/status_im/data_store/realm/core.cljs:
```diff
@@ -49,11 +49,9 @@
-    (reset-realm file-name schemas)
-    (migrate-realm file-name schemas)))
+  ;; install has incompatible database schemas) 
+  (reset-realm file-name schemas)
```
have you tested this ? To me it seems like it will just delete the db everytime you start the app ?

janherich replied:
You are correct

janherich commented:
@yenda I fixed the logic:
I added dummy `v2` schema versions to `release/0.9.18`, and added condition that realm purge will be performed whenever installed schema version is lower then highest schema version in codebase (which will be the case for `0.9.17` -> `0.9.18` install (`v1` -> `v2` change)).
From now on we should probably restore the good practice of introducing realm changes in migrations and start with **v3** (so skip **v2** in develop), as **v2** will be the schema version of the newest release (`0.9.18`) app.

yenda commented:
@janherich I think it would be cleaner to have a file written the first time when you reset.

janherich commented:
@yenda you mean some special "lock" file ?

yenda commented:
yes for instance v18.lock, reset if not present

janherich commented:
@yenda I don't particularly like the idea of some another persistance source other then realm, with it's own write/read API, even if it's just simple flat file.
The solution with Version jump may sound hacky (it is), but it's the simplest thing to do and once V3 migration lands in develop, we can completely forget about it.

antdanchenko commented:
An error is shown after upgrading 0.9.17 with [PR build](https://i.diawi.com/kVRZDR) on Android

![screenshot_2018-05-02-16-59-09](https://user-images.githubusercontent.com/12611990/39534831-2f86bef4-4e3b-11e8-9040-cb8fb4241f7f.png)

An error is shown after upgrading 0.9.17 with [PR build](https://i.diawi.com/MyUko2) on IOS

![image_uploaded_from_ios](https://user-images.githubusercontent.com/12611990/39535140-d79d470c-4e3b-11e8-9ba8-8312730f3b66.png)

After pressing 'OK' a white screen is shown for both platforms

**Steps to reproduce:**
Install 0.9.17 
Create an account on 0.9.17 version
Upgrade 0.9.17 with the PR build

antdanchenko commented:
### Actual Result:
App hangs on `Signing you in..` after updating 0.9.17 to 0.9.18 '3451e9' (android only)

![image](https://user-images.githubusercontent.com/12611990/39633368-6f27a7c2-4fc0-11e8-85d3-f231a9277ace.png)

### Steps to Reproduce:

- Install 0.9.17 from PlayMarket and create an account 
- Upgrade 0.9.17 to 0.9.18 '3451e9
- Open the app and enter valid password and tap on sign in

@janherich It won't be easy to introduce an upgrade in 0.9.18 release, also with https://github.com/status-im/ideas/tree/master/ideas/142-recovery-compatibility we will exclude it one more time in 0.9.19 or 0.9.20

It seems that the best way is to implement `Expected behavior` from https://github.com/status-im/status-react/issues/4077 (An alert about excluded upgrade option from the release is shown, no crashes, user can create new account or recover the old one)

<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/ideas">status-im/ideas</a></strong></div><div>ideas - Swarm Home. New, completed and in-progress features for Status</div></blockquote>
